Plugins: Add a filter bar to switch between business models

ryelle opened the pull request:
This adds "Commercial" and "Community" (and "All") links to the top of the plugins grid. When clicked, it filters the current archive by the selected business model.

This taxonomy already existed as "Business model". The filter URLs already work on production: for example, [all community plugins](https://wordpress.org/plugins/?plugin_business_model=community) or [commercial block directory plugins](https://wordpress.org/plugins/browse/block/?plugin_business_model=commercial).

When the archive title is generated, it's "Business Model: Commercial" & "Business Model: Community" — do we want to change that at all?

The filters exist on all archives except the "beta" view, since these are likely all "community" (or eventually "canonical"?). Should they apply to "My Favorites" (see screenshot below)?

The mockups also include a "Canonical" link, which does exist on the site, but there are no plugins in it yet, so I did not add it here.

cc @WordPress/meta-design 

| Before | After |
|---|---|
| ![before](https://github.com/WordPress/wordpress.org/assets/541093/fb265e60-796f-4743-aaa0-bd4465196e83) | ![home](https://github.com/WordPress/wordpress.org/assets/541093/fdefcd06-b74f-4137-99e6-66ab150537a9) |

All community plugins (https://wordpress.org/plugins/?plugin_business_model=community)
![filtered](https://github.com/WordPress/wordpress.org/assets/541093/43d58075-f287-4c89-a1f6-ed0a566adb28)

My Favorites — currently has the filters, but we could remove if this seems unnecessary. I removed the curation filters from favorites on [the Pattern Directory](https://github.com/WordPress/pattern-directory/pull/584), but that is a slightly different use case.

![favorites](https://github.com/WordPress/wordpress.org/assets/541093/549995f1-6fcd-4327-a4e9-ad31e70b83a7)

The filters also apply to rosetta sites:

| Arabic | Korean |
|---|---|
| ![rosetta-ar](https://github.com/WordPress/wordpress.org/assets/541093/dd2cb3d8-bd67-4179-b89b-41993265e16d) | ![rosetta-ko](https://github.com/WordPress/wordpress.org/assets/541093/845e04da-2477-450b-88eb-6be92c7b0949) |

Ticket: https://meta.trac.wordpress.org/ticket/7035

ryelle committed:
Plugins: Add a filter bar to switch between business models

StevenDufresne commented on wordpress.org/public_html/wp-content/themes/pub/wporg-plugins/header.php:
```diff
@@ -19,7 +19,21 @@
+global $wp_query;
+$is_beta = 'beta' === $wp_query->get( 'browse' );
+// The filter bar should not be shown on:
+// - search: wp-query filters not compatible with ES search.
```
This is unfortunate. I would expect it to be most valuable on the search page since browsing the directory isn't a great experience. 

During the redesign, we should make sure to get this working or at least scoped.

ryelle replied:
It is _possible_ to add it, assuming the value is synced to the elasticsearch database (i'm not sure where to confirm this), it would need to be added in before the ES query is built ([here, I think](https://github.com/WordPress/wordpress.org/blob/0860b124ae32e0b65129abca6787383bfde4f270/wordpress.org/public_html/wp-content/plugins/plugin-directory/libs/site-search/jetpack-search.php#L444-L457)).

So we can add the filter to search too, on the redesign.

StevenDufresne replied:
Sounds good.

StevenDufresne commented:
> When the archive title is generated, it's "Business Model: Commercial" & "Business Model: Community" — do we want to change that at all?

Maybe we can go with:

- `Commercial Plugins`
- `Community Plugins`

We should also probably update the copy on the filtered view before launch. They have pluralization problems:

`This plugin is developed and supported by a community.` should be `These plugins are developed and supported by a community.`
| Screenshot |
| --- |
| <img width="555" alt="Screenshot 2023-06-07 at 10 49 16 AM" src="https://github.com/WordPress/wordpress.org/assets/1657336/dffd0e78-f63d-45fc-9339-67c51c45fbec"> |

StevenDufresne commented:
> The filters exist on all archives except the "beta" view, since these are likely all "community" (or eventually "canonical"?). Should they apply to "My Favorites" (see screenshot below)?

I don't think it's necessary for that view.

dd32 commented on wordpress.org/public_html/wp-content/themes/pub/wporg-plugins/header.php:
```diff
@@ -47,16 +61,50 @@
+					) {
+						$class = 'is-active';
+					}
+					$url = home_url( '/' );
+					if ( $slug ) {
+						$url = add_query_arg( array( 'plugin_business_model' => $slug ) );
+					} else if ( ! is_home() ) {
+						$url = remove_query_arg( 'plugin_business_model' );
+					}
```
This should be achievable just by using `get_term_link()` I'd have thought? We probably want to enable pretty-permalinks for these anyway, and/or use `/browse/community/`.

I'm happy to see the design/functionality launch as-is **before** that happens though, it can be resolved afterwards.

ryelle replied:
I went with this so that we could easily build up filtered queries like `https://wordpress.org/plugins/browse/block/?plugin_business_model=commercial`, because I was under the impression that these are filters, not just static links (@jasmussen can confirm).

jasmussen commented:
Looks good, thanks for working on this.

ryelle committed:
Remove filters from favorites page

ryelle committed:
Update the archive title & description

ryelle commented:
I've removed the filters from the "Favorites" page, and updated the title & description on the archive pages, so those now look like this:

| Community | Commercial |
|---|---|
| <img width="624" alt="Screenshot 2023-06-07 at 4 53 47 PM" src="https://github.com/WordPress/wordpress.org/assets/541093/2a0c4f51-4022-4113-b561-a0cc57f2daf0"> | <img width="554" alt="Screenshot 2023-06-07 at 4 54 08 PM" src="https://github.com/WordPress/wordpress.org/assets/541093/734264c5-3b8f-47bd-b3cf-177473d5b11f"> |

ryelle commented:
Committed in [r12643](https://meta.trac.wordpress.org/changeset/12643).
